openssl/src/x509/mod.rs

```diff
@@ -443,6 +460,153 @@ impl X509Ref {
+            let data = key_usage_str.as_slice();
+
+            for i in 0..num_bytes {
+                flags ^= (data[i] as u32) << ((num_bytes - (i + 1)) * 8);
+            }
+
+            println!("{}", num_bytes);
+
+            KeyUsageFlags::from_bits(flags)
+        }
+    }
+
+    /// Returns this certificate's extended key usage, if it exists.
+    ///
+    /// This corresponds to [`X509_get_ext_d2i`] called with `NID_ext_key_usage`.
+    ///
+    /// [`X509_get_ext_d2i`]: https://www.openssl.org/docs/man1.1.0/crypto/X509_get_ext_d2i.html
+    pub fn ext_key_usage(&self) -> Option<Stack<Asn1Object>> {
+        unsafe {
+            let ext_key_usage = ffi::X509_get_ext_d2i(
+                self.as_ptr(),
+                ffi::NID_ext_key_usage,
+                ptr::null_mut(),
+                ptr::null_mut(),
+            );
+
+            if ext_key_usage.is_null() {
+                None
+            } else {
+                Some(Stack::from_ptr(ext_key_usage as *mut _))
+            }
+        }
+    }
+
+    /// Returns this certificate `AuthorityKeyId`, if it exists.
+    ///
+    /// This corresponds to [`X509_get_ext_d2i`] called with `NID_authority_key_identifier`.
+    ///
+    /// [`X509_get_ext_d2i`]: https://www.openssl.org/docs/man1.1.0/crypto/X509_get_ext_d2i.html
+    pub fn authority_keyid(&self) -> Option<AuthorityKeyId> {
+        unsafe {
+            let akid = ffi::X509_get_ext_d2i(
+                self.as_ptr(),
+                ffi::NID_authority_key_identifier,
+                ptr::null_mut(),
+                ptr::null_mut(),
+            );
+
+            if akid.is_null() {
+                None
+            } else {
+                Some(AuthorityKeyId::from_ptr(akid as *mut _))
+            }
+        }
+    }
+
+    /// Returns this certificate subject key ID, if it exists.
+    ///
+    /// This corresponds to [`X509_get_ext_d2i`] called with `NID_subject_key_identifier`.
+    ///
+    /// [`X509_get_ext_d2i`]: https://www.openssl.org/docs/man1.1.0/crypto/X509_get_ext_d2i.html
+    pub fn subject_keyid(&self) -> Option<Asn1String> {
+        unsafe {
+            let skid = ffi::X509_get_ext_d2i(
+                self.as_ptr(),
+                ffi::NID_subject_key_identifier,
+                ptr::null_mut(),
+                ptr::null_mut(),
+            );
+
+            if skid.is_null() {
+                None
+            } else {
+                Some(Asn1String::from_ptr(skid as *mut _))
+            }
+        }
+    }
+
+    /// Returns an iterator that allows iterating over the `X509Extensions` present in
+    /// this certificate.
+    ///
+    /// This issues one call to [`X509_get_ext_count`] to get the number of extensions, while the
+    /// iterator itself will lazily fetch each extension.
+    ///
+    /// [`X509_get_ext_count`]: https://www.openssl.org/docs/man1.1.0/crypto/X509_get_ext_count.html
+    pub fn extensions(&self) -> Extensions {
+        let extension_count: c_int;
```

Comment:
You can simplify this to just `let exension_count = unsafe { ffi::X509_get_ext_count(self.as_ptr()) };`.